app/lib/activitypub/activity/create.rb

```diff
@@ -165,40 +165,77 @@ def in_reply_to_uri
   end
 
   def text_from_content
+    return Formatter.instance.linkify([text_from_name, object_url].join(' ')) if converted_object_type?
+
     if @object['content'].present?
       @object['content']
-    elsif language_map?
+    elsif content_language_map?
       @object['contentMap'].values.first
     end
   end
 
-  def language_from_content
-    return LanguageDetector.instance.detect(text_from_content, @account) unless language_map?
-    @object['contentMap'].keys.first
+  def text_from_name
+    if @object['name'].present?
+      @object['name']
+    elsif name_language_map?
+      @object['nameMap'].values.first
+    end
+  end
+
+  def detected_language
+    if content_language_map?
+      @object['contentMap'].keys.first
+    elsif name_language_map?
+      @object['nameMap'].keys.first
+    elsif supported_object_type?
```

Comment:
why not do this for converted object types?

Reply:
If a Video does not have a language attribute, what's the point of trying to detect language from its description? What would language even refer to if we're talking about a Video or an Image? for text posts it's clear. For media, it is not, because the thing itself is not the text, the text is merely a description, which may be missing, which may be in a different language.

Reply:
There may be other converted object types, like if we wanted to implement base Activities, but I see your point.

Reply:
>There may be other converted object types

AP doesn't have a general language attribute. It can only provide languages for name/content. That means it can only be relevant for text posts where object == its text.